support-models/src/main/scala/com/gu/support/workers/states/CreateZuoraSubscriptionState.scala

```diff
@@ -45,6 +46,8 @@ object CreateZuoraSubscriptionProductState {
       product: SupporterPlus,
       paymentMethod: PaymentMethod,
       salesForceContact: SalesforceContactRecord,
+      promoCode: Option[PromoCode],
+      user: User,
```

Comment:
This is fine to be required as it is sent from all endpoints.

Reply:
it is true that it's sent from all the endpoints but in the tests we don't want to bloat them too much - if you just pick only the billing country then it makes it easier from that side.

A big problem we have in tests in this code is you need a ridiclous amount of data passed in, half of which isn't actually needed for your specific test, hence the desire to minimise the state.  Blue peter style, there's one I made earlier here:
https://github.com/guardian/support-frontend/pull/5673/files#diff-1eed1787d655713c7200add111db3c6809de259f3b4731166f2303eb91811e64R45

I also notice you added the params on the end, which usually makes sense, but I actually tried to keep the ordering more similar to other of the CreateZuora*State classes, just to improve consistency.  Hence my billing country is first parameter.

Reply:
The problem I find is that we are essentially creating new transformation steps for all these types to narrow down their fields, making changes quite hard to work out where the transformation has happened.

It is also a load more code for each transformation. 

Test fixtures would be easy to create by just generating 1 full version and the modifying the fields we need to to make the test make sense.

Happy to chat about this offline as I have found this issue get me a few times and worry it will for anyone new to the codebase.